Src/AwesomeAssertions/Primitives/StringEqualityStrategy.cs

```diff
@@ -18,110 +25,64 @@ public StringEqualityStrategy(IEqualityComparer<string> comparer, string predica
     {
-        ValidateAgainstSuperfluousWhitespace(assertionChain, subject, expected);
-
-        if (expected.IsLongOrMultiline() || subject.IsLongOrMultiline())
-        {
-            int indexOfMismatch = subject.IndexOfFirstMismatch(expected, comparer);
-
-            if (indexOfMismatch == -1)
-            {
-                ValidateAgainstLengthDifferences(assertionChain, subject, expected);
-                return;
-            }
-
-            string locationDescription = $"at index {indexOfMismatch}";
-            var matchingString = subject[..indexOfMismatch];
-            int lineNumber = matchingString.Count(c => c == '\n');
-
-            if (lineNumber > 0)
-            {
-                var indexOfLastNewlineBeforeMismatch = matchingString.LastIndexOf('\n');
-                var column = matchingString.Length - indexOfLastNewlineBeforeMismatch;
-                locationDescription = $"on line {lineNumber + 1} and column {column} (index {indexOfMismatch})";
-            }
-
-            string mismatchSegment = GetMismatchSegmentForLongStrings(subject, expected, indexOfMismatch).EscapePlaceholders();
-
-            assertionChain.FailWith($$"""
-                {{ExpectationDescription}}the same string{reason}, but they differ {{locationDescription}}:
-                {{mismatchSegment}}.
-                """);
-        }
-        else if (ValidateAgainstLengthDifferences(assertionChain, subject, expected))
+        if (ValidateAgainstSuperfluousWhitespace(assertionChain, subject, expected))
         {
-            int indexOfMismatch = subject.IndexOfFirstMismatch(expected, comparer);
-
-            if (indexOfMismatch != -1)
-            {
-                assertionChain.FailWith(
-                    $"{ExpectationDescription}{{0}}{{reason}}, but {{1}} differs near " + subject.IndexedSegmentAt(indexOfMismatch) +
-                    ".",
-                    expected, subject);
-            }
+            return;
         }
-    }
 
-    public string ExpectationDescription => $"Expected {{context:string}} to {predicateDescription} ";
+        int indexOfMismatch = GetIndexOfFirstMismatch(subject, expected);
 
-    private void ValidateAgainstSuperfluousWhitespace(AssertionChain assertion, string subject, string expected)
-    {
-        assertion
-            .ForCondition(!(expected.Length > subject.Length && comparer.Equals(expected.TrimEnd(), subject)))
-            .FailWith($"{ExpectationDescription}{{0}}{{reason}}, but it misses some extra whitespace at the end.", expected)
-            .Then
-            .ForCondition(!(subject.Length > expected.Length && comparer.Equals(subject.TrimEnd(), expected)))
-            .FailWith($"{ExpectationDescription}{{0}}{{reason}}, but it has unexpected whitespace at the end.", expected);
+        assertionChain
+            .ForCondition(indexOfMismatch == -1)
+            .FailWith(() => new FailReason(CreateFailureMessage(subject, expected, indexOfMismatch)));
```

Comment:
I would do that lazy like so `.FailWith(() => ....);` how you did it in other assertions